### ISSUE TITLE
Vision Camera is great but is it worth it for Production apps ?

### ISSUE THREAD

**Issue by ratz6** [cuvent/react-native-vision-camera#1557]
### Question

Vision Camera is great but is it worth it for Production apps ? Seeing IG, Youtube and other short content apps everyone is having a great options of things on camera like insane filters and much more.. which is most of the time out of scope for this library. 
Not that I'm against Vision Camera but facts just say that this is not a production ready camera (yet). 

So any suggestions for any alternatives ? 
I might switch to Banuba's Video Editor SDK , but looking for other options as well

### VisionCamera Version

2.14.1

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for similar questions [in the issues page](https://github.com/mrousavy/react-native-vision-camera/issues?q=is%3Aopen+is%3Aissue+label%3A%22❓+question%22) as well as [in the discussions page](https://github.com/mrousavy/react-native-vision-camera/discussions) and found none.

**Comment by idrissakhi** [comment]
POV I don't think it's really worth it to use on Prod apps, since the repo owner does not allow a lot of maintainers, and just look for sponsors. I don't know why community have deprecated previous camera repo for this one, so finally we find our selves in such issue.

**Comment by rfischer20** [comment]
This is a tough question and probably not well suited the issues section.

In a simple manner - yes it is production ready. We use it in a rather large production app and have always been impressed with the work. It's built to serve as a camera with the capability to enhance with things you are asking.

**Comment by ratz6** [comment]
@rfischer20 can you name few apps which uses it in production ? Just want to see how complex have they gone with Vision Camera ..

I agree it's serve as a good basic camera , but still look at the market out there .. IG/YT and many more small apps have wonderful camera built in with crazy filters and multiple camera supports and every content creator is using such kind of apps so the expectations are high now .. but this library is not meeting those yet is what my point is, so the question .. are there any alternatives like Banuba VE ?  

Wonder if V3 is going to be worth the hype 

cc @mrousavy

**Comment by mrousavy** [comment]
> but facts just say that this is not a production ready camera (yet)

who says that? lol

**Comment by ratz6** [comment]
Product Managers 🥲
After Comparing Cameras of other apps

**Comment by mrousavy** [comment]
your product managers have no idea what they're talking about.

VisionCamera is fully production ready and used by many apps out there. I can't say all names, but for example [Unbogify](https://unbogify.com) does some insane Frame Processor magic with VisionCamera, and [Locket](https://apps.apple.com/us/app/locket-widget/id1600525061) used it as well.

What you're trying to say is that you need some advanced features like filters and multi-camera support.
I assume with multi-camera support you mean recording from front and back camera at the same time. That'd be possible with some changes to VisionCamera, but since I don't need that in my apps and no one pays me to build that, it's not there yet.
Filters will be possible in V3, although they are not final there. I agree that filters are not production ready yet.

Those advanced features are highly customized to an app's need, and if you ever designed a library you know it's almost impossible to build this into a general purpose camera library without significantly compromising on simpler more standard use-cases. For example, if VisionCamera spins up an AR session to add easy support for 3D filters, it will take longer to build and longer to start up for users that don't even use those features and just want a Camera.
With V3 I am trying to create an API that does not do any compromises but still delivers in terms of 2D filters.

If you want such advanced features developed specifically for your app, you can contact us at hello@margelo.io. We build such things almost on a daily basis, check out the Stori app we showcase on [our website](https://margelo.io) as an example.

**Comment by ratz6** [comment]
The apps which you have mentioned are pretty basic and 1 is without Video, exactly my point being that. Vision camera is great for such apps, but can we build something like tiktok, IG , etc with Vision Camera ? Any examples for that ?

We managed to fix Front and Back Camera toggling. So instead of having a single recording, we went ahead with clip approach like how YT and IG does and the user can change the camera before starting a new clip, and at the end of the recording we merge all clips using FFMPEG..

But any help with filters ? Even in V3 it will just be the start of filters right if it's there ? and when will it be there ? I want to sponsor so let me know the process how it works.. 

See my point of raising this issue was simply to ask for alternatives like Vision Camera but which is production ready for advanced apps. Nothing personal @mrousavy

**Comment by m1dok** [comment]
what do you mean by production?

Perhaps you mean it's not ready _for your production app._

Ig, tiktok ect are not libraries, and if you want a library that works like ig/tiktok cameras you should consider paying for a "specific" library for such a thing.

in my opinion you are wrong to compare two things that are **totally different.**

**Comment by ratz6** [comment]
@m1dok exactly that's what my question was.. thanks for taking your time and commenting.. you got it..
Whom should I pay ? Who has Video camera filters in react native ?

**Comment by m1dok** [comment]
@ratz6    i'm sure a google search will help, if you don't want a complete ready package like banuba or DeepAR, you can always find a freelance developer.

**Comment by myselfuser1** [comment]
Try this https://www.youtube.com/playlist?list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i

**Comment by mrousavy** [comment]
Hey, yes.
